app/assets/src/components/views/compare/SamplesHeatmapView/SamplesHeatmapControls.jsx

```diff
@@ -50,7 +50,7 @@ export default class SamplesHeatmapControls extends React.Component {
         value={this.props.selectedOptions.species}
         onChange={this.onTaxonLevelChange}
         label="Taxon Level"
-        disabled={!this.props.data}
+        disabled={this.props.loading}
```

Comment:
Is checking loading by itself better than checking if data is empty? Should we use both?

Reply:
I've added back `data` so that we check both since that seems to be a pattern elsewhere in the code as well.

Reply:
We should not just follow a pattern, but ask ourselves does it make sense for these filter to be enabled.
Data makes sense because they are useless without data, and we would have to consider that edge case, loading could make sense because we also do not want people changing filters while we are about to change data, which avoids plenty of other edge cases.

Reply:
I think having both makes sense in order to cover the most number of edge cases!

I believe we **need** to disable filters when the heatmap is in a loading state, otherwise filters can be applied when the data is in the middle of being fetched or in an incomplete state. And as you mentioned, having the filters disabled when there's no data makes sense.